db.go

```diff
@@ -32,6 +32,17 @@ func New(url string) (*DB, error) {
 	return &DB{ws}, nil
 }
 
+// NewWithTimeout Creates a new DB instance given a WebSocket URL and a timeout.
+// The timeout is in seconds.
+// The timeout is used for both the initial connection and for all subsequent requests.
+func NewWithTimeout(url string, timeout float64) (*DB, error) {
```

Comment:
I agree with that but I wanted the keep code smaller if possible.
We can discuss this.
if we will go in that direction that is the case I also would like to pass Time.Duration as a Parameter instead of float64

Reply:
NewWithOptions added.

Reply:
We can reduce all functions to just `New(url string, ... Option)` as mentioned above I think

Reply:
I think we may also want to move the option type and options to the main package as well

I dont think people will be able to access the type or the options because websocket is an internal package.
This would also make more sense if in the future we want to add options that aren't related to the websocket package

Reply:
The above conversation is resolved as @ElecTwix has reconciled exposing types and sharing a common baseline. The function signatures have also been reduced to a single one. Great discussion, both!